src/mapeo-project.js

```diff
@@ -626,9 +628,28 @@ export class MapeoProject extends TypedEmitter {
       ])
     )
 
-    // TODO: 3. Clear data from indexes
+    // 3. Clear data from indexes
     // 3.1 Reset multi-core indexer state
+    await Promise.all(
+      Object.values(this.#dataStores)
+        .filter((dataStore) => dataStore.namespace !== 'auth')
+        .map(async (dataStore) => {
+          await dataStore.close()
+          await dataStore.unlink()
+        })
+    )
+
     // 3.2 Clear indexed data
+    const isIndexedSchema = setHas(new Set(this.#indexWriter.schemas))
+    await Promise.all(
+      Object.values(this.#dataTypes)
+        .filter((dataType) =>
+          namespacesWithoutAuth.includes(dataType.namespace)
+        )
+        .map((dataType) => dataType.schemaName)
+        .filter(isIndexedSchema)
+        .map((schemaName) => this.#indexWriter.deleteSchema(schemaName))
+    )
 
     // 4. Assign LEFT role for device
     await this.#roles.assignRole(this.#deviceId, LEFT_ROLE_ID)
```

Comment:
I don't think this should work after the core is closed (line 625) and I don't think it should resolve after the indexer is closed.

Reply:
Moved in ab4d9c1194a1b286ce52d85a9f25f988857cb8c2.